garrysmod/lua/includes/extensions/string.lua

```diff
@@ -1,92 +1,76 @@
+local error = error
+local isnumber = isnumber
+local tonumber = tonumber
+local tostring = tostring
 
-local string = string
-local math = math
+local table_concat = table.concat
 
---[[---------------------------------------------------------
-	Name: string.ToTable( string )
------------------------------------------------------------]]
-function string.ToTable( str )
+local math_floor = math.floor
+local math_Round = math.Round
+
+local string_sub = string.sub
+local string_find = string.find
+local string_gsub = string.gsub
+local string_match = string.match
+local string_format = string.format
+
+getmetatable( "" ).__index = function( str, key )
+	if ( isnumber( key ) ) then
+		return string_sub( str, key, key )
+	end
+
+	local val = string[ key ]

```

Comment:
This should be checked first, we prioritize values from string library rather than operator[int].
Plus for faster lookup

Reply:
It's intentional in your case and I agree with that

Reply:
I think the ability to alter string[1] is a too big of a break in abstraction to ignore. Faster access yes but if a dev really cared about saving cycles, they'd localise the funcs anyway.